### PR TITLE
Heat 223 for fetching IPAllow List

### PR DESCRIPTION
**Pull request by dibyanshuMoj**

### CODE REVIEW

File: github_discovery.py [hunk 1]
```diff
@@ -249,8 +290,22 @@ def process_repo(**component):
       if file.name.startswith('values-'):
         env = re.match('values-([a-z0-9-]+)\\.y[a]?ml', file.name)[1]
         helm_environments.append(env)
+
+        # HEAT-223 Start : Read and collate data for allowlist from all environment specific values.yaml files.
+        ip_allow_list[file] = fetch_values_for_allowlist_key(get_file_yaml(repo, f"{monorepo_dir_suffix}helm_deploy/{file.name}"), allow_list_key)
```
Comment (mattops) [inline github_discovery.py:295]: this is duplicating code on line 299. Suggest you move this block after 299. and do 
```suggestion
        ip_allow_list[file] = fetch_values_for_allowlist_key(helm_default_values, allow_list_key) 
```

File: github_discovery.py [hunk 2]
```diff
@@ -249,8 +290,22 @@ def process_repo(**component):
...
+        ip_allow_list[file] = fetch_values_for_allowlist_key(get_file_yaml(repo, f"{monorepo_dir_suffix}helm_deploy/{file.name}"), allow_list_key) 
+        ip_allow_list_data.update({f"ip_allow_list_{env}": ip_allow_list[file]})
+        # HEAT-223 End : Read and collate data for allowlist from all environment specific values.yaml files.
+
     helm_default_values = get_file_yaml(repo, f"{monorepo_dir_suffix}helm_deploy/{c_name}/values.yaml")
+    
     if helm_default_values:
+
+      # HEAT-223 Start : Read and collate data for allowlist from default values.yaml files.   
+      default_values = "default"
```
Comment (mattops) [inline github_discovery.py:304]: suggest that default - is not specific or helpful, use `values.yaml` the name of the file as the key in your data.

File: github_discovery.py [hunk 3]
```diff
@@ -249,8 +290,22 @@ def process_repo(**component):
...
+      ip_allow_list[default_values] = fetch_values_for_allowlist_key(helm_default_values, allow_list_key)
+      ip_allow_list_data.update({f"ip_allow_list_{default_values}": ip_allow_list[default_values]})
```
Comment (mattops) [inline github_discovery.py:306]: maybe just do:
```suggestion
      ip_allow_list_data.update({"values.yaml": ip_allow_list[default_values]}) 
```

File: github_discovery.py [hunk 4]
```diff
@@ -249,8 +290,22 @@ def process_repo(**component):
       if file.name.startswith('values-'):
         env = re.match('values-([a-z0-9-]+)\\.y[a]?ml', file.name)[1]
         helm_environments.append(env)
+
+        # HEAT-223 Start : Read and collate data for allowlist from all environment specific values.yaml files.
+        ip_allow_list[file] = fetch_values_for_allowlist_key(get_file_yaml(repo, f"{monorepo_dir_suffix}helm_deploy/{file.name}"), allow_list_key) 
+        ip_allow_list_data.update({f"ip_allow_list_{env}": ip_allow_list[file]})
```
Comment (mattops) [inline github_discovery.py:296]:
```suggestion
        ip_allow_list_data.update({file.name: ip_allow_list[file]})
```

File: github_discovery.py (hunk 2 above)
Comment (dibyanshuMoj) [reply github_discovery.py:304]: Code on 295 is reading all values files except the default one and fetch the payload. Code after 299 is under if condition is reading default values file only if there is one. This is the existing way that is implemented to read default values file and other environment specific files.

Comment (dibyanshuMoj) [reply github_discovery.py:303]: Done.

File: github_discovery.py (hunk 4 above)
Comment (dibyanshuMoj) [reply github_discovery.py:295]: done

Comment (mattops) [reply github_discovery.py:295]: is it done?

File: github_discovery.py (hunk 3 above)
Comment (mattops) [reply github_discovery.py:305]: still needs updating??

**Pull request closed without merging**